src/js/sjcl-custom/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+/*global sjcl*/
+/*eslint-disable new-cap*/
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function (hash, hashObject, q) {
+
+  hashObject = hashObject || sjcl.hash.sha256;
+  var secret = this.get();
+
+  /* k = 0x00 0x00 0x00 ... 0x00 */
+  var k = [];
+  while (sjcl.bitArray.bitLength(k) < sjcl.bitArray.bitLength(hash)) {
+    k = sjcl.bitArray._shiftRight(k, 8);
```

Comment:
We should avoid using private methods starting with `_`. I think you can do this with `sjcl.bn`.